### PR TITLE
Add warning when user use Object for selectedItem

### PR DESCRIPTION
**Pull request by philipyoungg**
**What**: Add `console.warn` when user passed Object to `selectedItem`

**Why**: To improve DX—to minimize a case when user get `[object Object]` as an input value. Almost refactored my whole codebase. I thought I can't use `Object` as a selected value. Thankfully I found `itemToString` API that accept a function with current `selectedItem` passed on the argument.

**How**: Two things: 
1. Adding simple `isPlainObject` function that check if the `selectedItem` is Object literal, and
2. use that function inside `itemToString`'s defaultProps

**Checklist**:

- [N/A] Documentation
- [x] Tests
- [x] Ready to be merged 
- [x] Added myself to contributors table

### CODE REVIEW

File: src/downshift.js [hunk 1]
```diff
@@ -67,7 +68,12 @@ class Downshift extends Component {
     defaultIsOpen: false,
     getA11yStatusMessage,
     id: generateId('downshift'),
-    itemToString: i => (i == null ? '' : String(i)),
+    itemToString: i => {
+      if (isPlainObject(i)) {
```
Comment (kentcdodds) [inline src/downshift.js]: I'd prefer this to only happen in dev. I think the build is wired up to handle `process.env.NODE_ENV` properly. Could you make this: `if (process.env.NODE_ENV !== 'production' && isPlainObject(i)) {`

File: src/downshift.js [hunk 2]
```diff
@@ -67,7 +68,12 @@ class Downshift extends Component {
...
+        console.warn('you passed an Object as the value of `selectedItem`. Please refer to `itemToString` API documentation on Downshift\'s repository.')
```
Comment (kentcdodds) [inline src/downshift.js]: I think prettier should have reformatted this automatically when you committed. Did you `npm install`?

File: src/downshift.js [hunk 3]
```diff
@@ -67,7 +68,12 @@ class Downshift extends Component {
     defaultIsOpen: false,
     getA11yStatusMessage,
     id: generateId('downshift'),
-    itemToString: i => (i == null ? '' : String(i)),
+    itemToString: i => {
```
Comment (kentcdodds) [inline src/downshift.js]: I'd like to have a test for this. Could you add one in [`downshift.misc.js`](https://github.com/paypal/downshift/blob/master/src/__tests__/downshift.misc.js)? Just do `jest.spyOn(console, 'warn')` on the first line of the test, then `console.warn.mockRestore()` on the last line to clean up. You could even use a snapshot if you want: `expect(console.warn).toHaveBeenCalledTimes(1)` and then `expect(console.warn.mock.calls).toMatchSnapshot()`

Comment (philipyoungg) [reply src/downshift.js]: Sure!

File: src/downshift.js (hunk 1 above)
Comment (philipyoungg) [reply src/downshift.js]: Would `process.env.NODE_ENV === 'development'` more prefereable in this instance?

**Pull request merged**